lib/scsi_defs.h

```diff
@@ -195,6 +195,100 @@ enum class PageCode : uint8_t {
+// SCSI Reference Manual Table 441
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+enum class ProductType : uint8_t {
+  kNotIndicated = 0x00,
+  kCFast = 0x01,
+  kCompactFlash = 0x02,
+  kMemoryStick = 0x03,
+  kMultiMediaCard = 0x04,
+  kSecureDigitalCard = 0x05,
+  kXQD = 0x06,
+  kUniversalFlashStorage = 0x07
+};
+
+// SCSI Reference Manual Table 442
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+enum class Wacereq : uint8_t {
+  kNotSpecified = 0b00,
+
+  // The device server completes the read command specifying that LBA with GOOD
+  // status and any data transferred to the Data-In Buffer is indeterminate.
+  kSuccess = 0b01,
+
+  // The device server terminates the read command specifying that LBA with
+  // CHECK CONDITION status with sense key set to MEDIUM ERROR and the
+  // additional sense code set to an appropriate value other than WRITE AFTER
+  // SANITIZE REQUIRED (e.g., ID CRC OR ECC ERROR).
+  kFailureWithoutWriteSanitize = 0b10,
+
+  // The device server terminates the read command specifying that LBA with
+  // CHECK CONDITION status with sense key set to MEDIUM ERROR and the
+  // additional sense code set to WRITE AFTER SANITIZE REQUIRED.
+  kFailureWithWriteSanitize = 0b11
+};
+
+// SCSI Reference Manual Table 443
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+enum class Wabereq : uint8_t {
+  kNotSpecified = 0b00,
+
+  // The device server completes the read command specifying that LBA with GOOD
+  // status and any data transferred to the Data-In Buffer is indeterminate.
+  kSuccess = 0b01,
+
+  // The device server terminates the read command specifying that LBA with
+  // CHECK CONDITION status with sense key set to MEDIUM ERROR and the
+  // additional sense code set to an appropriate value other than WRITE AFTER
+  // SANITIZE REQUIRED (e.g., ID CRC OR ECC ERROR).
+  kFailureWithoutWriteSanitize = 0b10,
+
+  // The device server terminates the read command specifying that LBA with
+  // CHECK CONDITION status with sense key set to MEDIUM ERROR and the
+  // additional sense code set to WRITE AFTER SANITIZE REQUIRED.
+  kFailureWithWriteSanitize = 0b11
+};
+
+// SCSI Reference Manual Table 444
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+enum class NominalFormFactor : uint8_t {
+  kNotReported = 0x0,
+  kExtraLarge = 0x1,  // 5.25 inch
+  kLarge = 0x2,       // 3.5 inch
+  kMedium = 0x3,      // 2.5 inch
+  kSmall = 0x4,       // 1.8 inch
+  kExtraSmall = 0x5,  // less than 1.8 inch
+};
+
+// SCSI Reference Manual Table 445
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+enum class Zoned : uint8_t {
+  kNotReported = 0b00,
+
+  // Device server implements the host aware zoned block device capabilities
+  // defined in ZBC
+  kHostAware = 0b01,
+
```

Comment:
nit: personally, I feel omitting the newlines here (and elsewhere) makes this struct more readable